libs/components/forms/src/lib/modules/radio/radio-group.component.ts

```diff
@@ -212,69 +230,69 @@ export class SkyRadioGroupComponent
   }
 
   public registerOnChange(fn: (value: any) => void): void {
-    this.onChange = fn;
+    this.#onChange = fn;
   }
 
   public registerOnTouched(fn: any): void {
-    this.onTouched = fn;
+    this.#onTouched = fn;
   }
 
   // eslint-disable-next-line @typescript-eslint/no-empty-function
-  private onChange: (value: any) => void = () => {};
+  #onChange: (value: any) => void = () => {};
   // eslint-disable-next-line @typescript-eslint/no-empty-function
-  private onTouched: () => any = () => {};
+  #onTouched: () => any = () => {};
 
-  private updateRadioButtonDisabled(): void {
+  #updateRadioButtonDisabled(): void {
     if (this.radios) {
       this.radios.forEach((radio) =>
         radio.setGroupDisabledState(this.disabled)
       );
     }
   }
 
-  private updateRadioButtonNames(): void {
+  #updateRadioButtonNames(): void {
     if (this.radios) {
       this.radios.forEach((radio) => {
         radio.name = this.name;
       });
     }
   }
 
-  private updateRadioButtonTabIndexes(): void {
+  #updateRadioButtonTabIndexes(): void {
     if (this.radios) {
       this.radios.forEach((radio) => {
         radio.groupTabIndex = this.tabIndex;
       });
     }
   }
 
-  private updateCheckedRadioFromValue(): void {
+  #updateCheckedRadioFromValue(): void {
     if (!this.radios) {
       return;
     }
 
     this.radios.forEach((radio) => {
-      radio.checked = this._value === radio.value;
+      radio.checked = this.#_value === radio.value;
     });
   }
 
-  private resetRadioButtons(): void {
-    this.updateCheckedRadioFromValue();
-    this.updateRadioButtonNames();
-    this.updateRadioButtonTabIndexes();
-    this.updateRadioButtonDisabled();
+  #resetRadioButtons(): void {
+    this.#updateCheckedRadioFromValue();
+    this.#updateRadioButtonNames();
+    this.#updateRadioButtonTabIndexes();
+    this.#updateRadioButtonDisabled();
   }
 
   #updateValue(value: any, markDirty: boolean): void {
-    const isNewValue = value !== this._value;
+    const isNewValue = value !== this.#_value;
 
     /* istanbul ignore else */
     if (isNewValue) {
-      this._value = value;
+      this.#_value = value;
```

Comment:
Accessing a setter's property outside of the setter?

Reply:
Ya... I don't love it either. The trick here is the difference between if we mark the form dirty or not. The setter and `writeValue` both call this function as their logic with a boolean flag. There is no way to pass that boolean flag to the setter and so we would still need to set the underscore variable in `writeValue` even if this function didn't exist.

Reply:
I think we've uncovered some "smelly" code. You're right that there's two entry points into the form control, one for the user's input (the `value` input), and one for the control's programmatic updates (`writeValue`). We might need to rethink how we're handling the "true" value of the control, and instead of accessing the `#_value` field here, we should create a new private property called `#controlValue` (or something) that both the `value` input and the `writeValue` method update (`#_value` would only represent the value passed to the `value` input).

Historically, we haven't been the best about separating the concerns of user input versus `writeValue` calls, so worst case, I think it would be better to duplicate code to represent the two concerns (if needed), rather than have branching logic in a singular method that both the `value` input and `writeValue` method call into.